src/pybind/mgr/volumes/fs/subvolume.py

```diff
@@ -189,6 +189,28 @@ def get_subvolume_path(self, spec):
             raise VolumeException(-e.args[0], e.args[1])
         return path
 
+    def get_dir_entries(self, path):
+        """
+        Get the directory names in a given path
+        :param path: the given path
+        :return: the list of directory names in json format
+        """
+        try:
+            dir_handle = self.fs.opendir(path)
+        except cephfs.ObjectNotFound:
+            raise VolumeException(-errno.ENOENT,
+                                  "Path '{0}' not found".format(path))
+        except cephfs.Error as e:
+            raise VolumeException(-e.args[0], e.args[1])
+
+        dirs = []
+        d = self.fs.readdir(dir_handle)
```

Comment:
a more canonical way to do this is:

````
with dir_handle:
    ...
````

Then you don't need to close the handle and exceptions close the directory immediately. (See also cephfs-shell.)